### PR TITLE
Add string.findall APIs to pylibcudf

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
Contributes to https://github.com/rapidsai/cudf/issues/15162

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/pylibcudf/pylibcudf/strings/findall.pyx
Comment (wence-) [inline python/pylibcudf/pylibcudf/strings/findall.pyx:14]: Can we call this argument `pattern` (rather than `prog`). It's kind of a leaky implementation detail that the type is called `Program`?

**Review by wence- — APPROVED** [review]
Tiny doc nit

**Comment by wence-** [comment]
I set to do not merge until https://github.com/rapidsai/cudf/pull/16813 is in, so we don't keep getting merge conflicts there. Once #16813, this is safe to merge.

**Review by Matt711 — APPROVED** [review]
Should be able to merge when CI is unblocked

**Pull request merged**
